integration/v3_grpc_test.go

```diff
@@ -193,3 +195,225 @@ func TestV3DeleteRange(t *testing.T) {
 		clus.Terminate(t)
 	}
 }
+
+// TestV3WatchFromCurrentRevision tests Watch APIs from current revision.
+func TestV3WatchFromCurrentRevision(t *testing.T) {
+	tests := []struct {
+		initialKeys   []string
+		initialValue  string
+		watchRequests []pb.WatchRequest
+		events        []storagepb.Event
+	}{
+		// watch the key
+		{
+			[]string{"foo"},
+			"bar",
+			[]pb.WatchRequest{{CreateRequest: &pb.WatchCreateRequest{Key: []byte("foo")}}},
+			[]storagepb.Event{
+				{
+					Kv: &storagepb.KeyValue{Key: []byte("foo"), Value: []byte("bar"), CreateRevision: 1, ModRevision: 1, Version: 1},
+				},
+			},
+		},
+		// watch the prefix, matching
+		{
+			[]string{"foolong"},
+			"bar",
+			[]pb.WatchRequest{{CreateRequest: &pb.WatchCreateRequest{Prefix: []byte("foo")}}},
+			[]storagepb.Event{
+				{
+					Kv: &storagepb.KeyValue{Key: []byte("foolong"), Value: []byte("bar"), CreateRevision: 1, ModRevision: 1, Version: 1},
+				},
+			},
+		},
+		// watch the prefix, non-matching
+		{
+			[]string{"foo"},
+			"bar",
+			[]pb.WatchRequest{{CreateRequest: &pb.WatchCreateRequest{Prefix: []byte("foobar")}}},
+			[]storagepb.Event{},
+		},
+	}
+
+	for i, tt := range tests {
+		clus := newClusterGRPC(t, &clusterConfig{size: 3})
+
+		wAPI := pb.NewWatchClient(clus.RandConn())
+		wStream, err := wAPI.Watch(context.TODO())
+		if err != nil {
+			t.Errorf("#%d: wAPI.Watch error: %v", i, err)
+		}
+
+		for _, r := range tt.watchRequests {
+			if err := wStream.Send(&r); err != nil {
+				t.Errorf("#%d: wStream.Send error: %v", i, err)
+			}
+		}
+
+		kvc := pb.NewKVClient(clus.RandConn())
+		for _, k := range tt.initialKeys {
+			reqput := &pb.PutRequest{Key: []byte(k), Value: []byte(tt.initialValue)}
+			if _, err := kvc.Put(context.TODO(), reqput); err != nil {
+				t.Errorf("#%d: couldn't put key (%v)", i, err)
+			}
+		}
+
+		copied := []storagepb.Event{}
```

Comment:
`Recv` returns `[]*Event` where I define test cases with `[]Event`. So I made one same to another.